plugin/evm/vm.go

```diff
@@ -597,7 +597,6 @@ func (vm *VM) Shutdown() error {
 
 // buildBlock builds a block to be wrapped by ChainState
 func (vm *VM) buildBlock() (snowman.Block, error) {
-	vm.limitOrderProcesser.AddMatchingOrdersToTxPool()
```

Comment:
why remove this logic away from block production time and put it at parseTx? - duplicate matched order txs will be issued, no?

Reply:
Seems like you accidentally replied to both comments in a single comment: 
> For now i am not matching against best price as contract is not updated for that. Also partial matching is not being done now. We will need to make contract changes to do these things. When we do that we don't have to change GetOrdersByPriceAndPositionType method. We will just change our matching logic in limit_order.go. I will add a comment there if needed.

ok so `GetOrdersByPriceAndPositionType` intends to return all possible matches correct? I think then the equality in `getLongOrdersByPrice` should be `price <= order.Price` and same for `getShortOrdersByPrice`